_posts/2021-02-16-watch-your-dependencies.md

```diff
@@ -0,0 +1,57 @@
+---
+layout: post
+title: Beware of dependency confusion (and other dependency attacks)
+author: lokori
+excerpt: Do you know how exactly your build tool resolves your dependencies? Are you sure? Dependency confusion is yet another subtle attack that can hit you if you are not careful, but it's definitely not the only one.
+tags:
+- Supply chain attack
+- Dependency Confusion
+- BOM
+---
+
+Nowadays most software developers have set up some kind of dependency analysis to detect known security issues in outdated packages, using tools like [OWASP Dependency Check](https://owasp.org/www-project-dependency-check/), [retire.js](https://github.com/retirejs/retire.js/) and others, but **dependency confusion** is a new subtle type of attack. Is your project vulnerable to it? How can you know? Read on.
+
+**Dependency confusion** means that a package resolver erroneously downloads a private package from a public repository. Since that package doesn't exist, and build tools don't usually require signatures, anyone can create such a package in a public repository. Subtle, easy to exploit and simple. And quite effective! People have made over 100k in just a few days from bug bounty programs by pwning Google, Facebook and other big companies with this. And it's not over yet - fun & profit indeed.
```

Comment:
Since that package doesn't exist -> Since names of private packages are not reserved in public repositories

Or something along those lines.